pkg/controller/clusterinstallation/mysql.go

```diff
@@ -18,36 +18,38 @@ import (
 
 func (r *ReconcileClusterInstallation) checkMySQLCluster(mattermost *mattermostv1alpha1.ClusterInstallation, reqLogger logr.Logger) error {
 	reqLogger = reqLogger.WithValues("Reconcile", "mysql")
-	cluster := mattermostmysql.Cluster(mattermost)
+	desired := mattermostmysql.Cluster(mattermost)
 
-	err := r.createMySQLClusterIfNotExists(mattermost, cluster, reqLogger)
+	err := r.createMySQLClusterIfNotExists(mattermost, desired, reqLogger)
 	if err != nil {
 		return err
 	}
 
-	foundCluster := &mysqlOperator.MysqlCluster{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: cluster.Name, Namespace: cluster.Namespace}, foundCluster)
+	current := &mysqlOperator.MysqlCluster{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: desired.Name, Namespace: desired.Namespace}, current)
 	if err != nil {
 		return err
 	}
 
+	// Updating mysql.cluster with objectMatcher breaks mysql operator.
+	// Only fields that are expected to be changed by mattermost-operator should be included here.
```

Comment:
Until other problems around database resource are resolved, mysql reconcile connot be changed to objectMatcher.